core/capabilities/ccip/oraclecreator/plugin.go

```diff
@@ -53,12 +53,12 @@ import (
 )
 
 var _ cctypes.OracleCreator = &pluginOracleCreator{}
-var extraDataCodec = ccipcommon.NewExtraDataCodec()
+var extraDataCodec = ccipcommon.NewExtraDataCodec(ccipevm.ExtraDataDecoder{}, ccipsolana.ExtraDataDecoder{})
 var plugins = map[string]plugin{
 	chainsel.FamilyEVM: {
 		CommitPluginCodec:   ccipevm.NewCommitPluginCodecV1(),
 		ExecutePluginCodec:  ccipevm.NewExecutePluginCodecV1(),
```

Comment:
So EVM's ExecutePluginCodec and MessageHasher should also need access to extraDataCodec, similar to how solana is needing them right?

I think currently it is not there, because we haven't supported Solana -> EVM side codecs.
In EVM codec, I see we are decoding extraArgs and destinationGas directly, assuming ABI encoding, and not using the extraDataCodec.

Could we make the change here itself?

Reply:
As we discussed offline, the evm changes can happen in another PR